### DISCUSSION TITLE
How to use different validation rules for the same form relatively to the chosen action?

### DISCUSSION THREAD

**Discussion by PetrovaDaria** [react-hook-form/react-hook-form#5772] [Q&A]
I want to build form which has two options how to save it (and they placed on the same page). One option is to save as a draft and the other is to save completely. Draft has only one required field. And in complete form almost every field is required. So I need to use different validation rules. Any ideas how to do it? 
The only idea that came to me is on click on every button change some "global" variables that used in validation rules `{required: isRequired}` and only then apply react-hook-form `handleSubmit` to validate form.

**Comment by sunnyfeijen** [comment]
@PetrovaDaria I did find a working solution, which I've written a bit about in [this dev.to post](https://dev.to/feijens/react-hook-form-multiple-yup-validation-schemas-5a22)

I hope this helps you!
